### PR TITLE
Feat/accounts and portfolios

### PR DESCRIPTION
**Pull request by gabrielcastr0**
# Midaz Console Pull Request Checklist

- Implements the functionality to fetch accounts along with their associated portfolios, including accounts without portfolios and vice versa.
- Updates the optional and required fields in the create accounts payload.

## Pull Request Type

[//]: # 'Check the appropriate box for the type of pull request.'

- [x] UI
- [x] Core (Business Lógic)
- [ ] Security
- [ ] internationalization
- [ ] Pipeline
- [ ] Documentation

## Checklist

Please check each item after it's completed.

- [x] I have tested these changes locally.
- [ ] I have updated the documentation accordingly.
- [ ] I have added necessary comments to the code, especially in complex areas.
- [x] I have ensured that my changes adhere to the project's coding standards.
- [ ] I have checked for any potential security issues.
- [ ] I have ensured that all tests pass.
- [ ] I have updated the version appropriately (if applicable).
- [x] I have confirmed this code is ready for review.

## Additional Notes

[//]: # 'Add any additional notes, context, or explanation that could be helpful for reviewers.'

### CODE REVIEW

File: src/app/(routes)/ledgers/[id]/accounts/accounts-tab-content.tsx
Comment (paulofdev) [inline src/app/(routes)/ledgers/[id]/accounts/accounts-tab-content.tsx:1]: desmembrar esse arquivo

File: src/app/(routes)/ledgers/[id]/portfolios/portfolios-tab-content.tsx
Comment (paulofdev) [inline src/app/(routes)/ledgers/[id]/portfolios/portfolios-tab-content.tsx:1]: desmembrar esse arquivo como em accounts

**Review by paulofdev — APPROVED** [review]

File: src/core/application/use-cases/portfolios-with-accounts/fetch-portfolios-with-account-use-case.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    page: number
+  ): Promise<PaginationDto<any>> {
+    const portfoliosResult = await this.fetchAllPortfoliosRepository.fetchAll(
+      organizationId,
+      ledgerId,
+      limit,
+      page
+    )
+
+    const allAccountsResult = await this.fetchAllAccountsRepository.fetchAll(
```
Comment (caioaletroca) [inline src/core/application/use-cases/portfolios-with-accounts/fetch-portfolios-with-account-use-case.tsx:41]: This is not going to work on the long run without a portfolioId filter on the accounts listing endpoint on midaz.

**Comment by caioaletroca** [comment]
@gabrielcastr0  I noticed that intl keys changed but there isn't any translation files changes.
Please run the extract, and using this opportunity, run test, lint and build too to check if everything is fine.

File: src/types/accounts-type.ts [hunk 2]
```diff
@@ -4,18 +4,21 @@ export type AccountType = {
...
-  alias: string
-  type: string
-  entityId: string
+  alias?: string
+  type?: string
+  entityId?: string
   parentAccountId: string
-  portfolioId: string
+  portfolioId?: string | null
+  portfolio: {
```
Comment (caioaletroca) [inline src/types/accounts-type.ts:12]: Create a Portfolio type on another file, import it here and use:

``` ts
{
 //...
 portfolio: Pick<PortfolioType, "name">
 //...
}
```

**Review by caioaletroca — CHANGES_REQUESTED** [review]
Thanks for that overhaul!!!

**Review by caioaletroca — APPROVED** [review]

**Pull request merged**